patches/woochimp.0002.fix.php-notice-woochimp-checkout-empty-groups.patch

```diff
@@ -16,7 +16,7 @@ index 0fe680b..85e25e6 100644
          if ($already_subscribed === false || ($already_subscribed && ($this->opt['woochimp_hide_checkbox'] == '2' || ($this->opt['woochimp_hide_checkbox'] == '3' && !empty($groups['data']) && !empty($groups['html']))))) {
              echo $checkbox_block;
 -            echo $groups['html'];
-+            if (!empty($groups['html'])) {
++            if (!empty($groups) && !empty($groups['html'])) {
```

Comment:
I think you should just add `!empty($groups)` like the following:

```php
if ($already_subscribed === false || ($already_subscribed && ($this->opt['woochimp_hide_checkbox'] == '2' || ($this->opt['woochimp_hide_checkbox'] == '3' && !empty($groups) && !empty($groups['data']) && !empty($groups['html']))))) { 
  echo $checkbox_block;
  echo $groups['html'];
}
```
⛔️ Please test first.

Reply:
But in this case `echo $groups['html'];` is reached if for example `$already_subscribed` is false but `$groups` might still be empty.

Reply:
Good point!